FlowCrypt/Functionality/DataManager/Encrypted Storage/Version5SchemaMigration.swift

```diff
@@ -254,4 +260,38 @@ final class Version5SchemaMigration {
 
         newObject[propertyName] = object
     }
+
+    private func setKeypairObjectPrimaryKey(_ object: MigrationObject) throws {
+        guard
+            let userObject = object[Properties.Keypair.user.rawValue] as? Object,
+            let email = userObject[Properties.User.email.rawValue] as? String,
+            let primaryFingerprint = object[Properties.Keypair.primaryFingerprint.rawValue] as? String
+        else {
+            throw AppErr.unexpected("KeypairRealmObject primary key")
+        }
+
+        object[Properties.Keypair.primaryKey.rawValue] = primaryFingerprint + email
```

Comment:
this is string concatenation, not a composed index, right? If string concat, then `primaryFingerprint + "," + base64(email)`.

The comma is necessary so that id of `abcd,efgh@example.com` and `abcdef,gh@example.com` (or their base64 representations) don't conflict

Reply:
Sure, it is still in progress so any ideas are welcome. I like idea with base64 encoding, let's use it.